src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -513,6 +541,13 @@ internal static T[] DangerousGetUnderlyingArray<T>(this ImmutableArray<T> array)
         internal static ReadOnlySpan<T> AsSpan<T>(this ImmutableArray<T> array)
             => array.DangerousGetUnderlyingArray();
 
+        internal static ImmutableArray<T> DangerousCreateFromUnderlyingArray<T>(ref T[] array)
```

Comment:
>internal [](start = 8, length = 8)

The fact that this is `internal` makes it much less terrifying to me.

Reply:
Yeah, I would certainly not want this to be public API.